mgmtfn/dockplugin/netDriver.go

```diff
@@ -742,10 +742,26 @@ func createNetworkHelper(networkID string, tag string, IPv4Data, IPv6Data []driv
 	return nil
 }
 
-// deleteNetworkHelper removes the association between docker network and contiv network
+// deleteNetworkHelper removes the association between docker network
+// and contiv network. We have to remove docker network state before
+// remove network in contiv.
 func deleteNetworkHelper(networkID string) error {
+	dnet, err := docknet.FindDocknetByUUID(networkID)
+	if err == nil {
+		// delete the dnet oper state
+		err = docknet.DeleteDockNetState(dnet.TenantName, dnet.NetworkName, dnet.ServiceName)
+		if err != nil {
+			log.Errorf("Couldn't delete docknet for nwID %s: %s", networkID, err.Error())
```

Comment:
Here, I would do something like:

```go
msg := fmt.Sprintf("my %s", "error message")
log.Error(msg)
return errors.New(msg)
```

Return as much useful info to the caller as possible

Reply:
done